portfolio/src/main/webapp/script.js

```diff
@@ -75,3 +75,11 @@ function displayComments () {
     document.getElementById('comment-section').innerHTML = out;
   });
 }
+
+function deleteComments () {
+  const request = new Request('/delete-data', {method: 'POST'});
+  fetch(request).then((response) => {
+    document.getElementById('comment-section').classList.add('empty');
+    displayComments();
```

Comment:
I think you can call this just once at the end of the loop. JS is mono-threaded so it won't make any difference to the UI anyway.